### PR TITLE
Adds `clasp push --watch`

### LINKED ISSUE #52: clasp push —watch
Add an option to watch local file changes and clasp push on file change.

```sh
clasp push —-watch
```

Use: https://www.npmjs.com/package/node-watch
Something like:

```ts
const watch = require('node-watch');
getProjectSettings().then(({ scriptId, rootDir }: ProjectSettings) => {
  watch(rootDir, { recursive: true }, (e, name) => {
    console.log(`${name} changed. Pushing...`);
    // Push...
    console.log('Pushed');
  });
}
```

### PR DESCRIPTION
**Pull request by grant**
Fixes #52.

Implements a basic watch command. Caveats:
- Does not only push changed files.
- Does not check if the changed file was an ignored file.

## Demo

```
clasp push --watch
Watching for changed files...

Pushing files...
└─ appsscript.json
└─ make.gs
└─ make2.gs
└─ make5.gs
└─ test.gs
Pushed 5 files.

- Updated: make5.gs

Pushing files...
└─ appsscript.json
└─ make.gs
└─ make2.gs
└─ make5.gs
└─ test.gs
Pushed 5 files.

- Updated: test.gs

Pushing files...
└─ appsscript.json
└─ make.gs
└─ make2.gs
└─ make5.gs
└─ test.gs
Pushed 5 files.
^C
```

- [x] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [x] Appropriate changes to README are included in PR.

### CODE REVIEW

File: src/files.ts [hunk 1]
```diff
@@ -165,4 +175,42 @@ export async function fetchProject(scriptId: string, rootDir = '', versionNumber
...
+
+/**
+ * Pushes project files to script.google.com.
+ */
+export async function pushFiles() {
+  const { scriptId, rootDir } = await getProjectSettings();
+  if (!scriptId) return;
+    getProjectFiles(rootDir, (err, projectFiles, files) => {
+      if(err) {
+        console.log(err);
```
Comment (campionfellin) [inline src/files.ts]: `logError(err, 'message')`

File: src/files.ts [hunk 2]
```diff
@@ -165,4 +175,42 @@ export async function fetchProject(scriptId: string, rootDir = '', versionNumber
...
   });
+}
+
+/**
+ * Pushes project files to script.google.com.
+ */
+export async function pushFiles() {
+  const { scriptId, rootDir } = await getProjectSettings();
+  if (!scriptId) return;
+    getProjectFiles(rootDir, (err, projectFiles, files) => {
```
Comment (campionfellin) [inline src/files.ts]: Indentation

**Comment by campionfellin** [comment]
LGTM besides a couple small comments.

Questions:

Can we improve it to not push unchanged files? I'm worried about people with huge files that we keep having to push even if not changed.

What happens if the changed file is an ignored file? Does it fail to push then?

**Comment by grant** [comment]
Users currently just `clasp push` when they want to, and we push unchanged files. So we're not really creating a worse experience.

Besides, the API can handle a bit more stress than it has now.

This flag simply `clasp push`s on any file change in the project. It's the same as push, change ignored file, push.

File: src/files.ts (hunk 2 above)
Comment (grant) [reply src/files.ts]: Fixed.

**Comment by grant** [comment]
@campionfellin Yes, we did. I'm always facing build issues and this is a bad habit.
I created #227 to stop this habit.

**Pull request merged**